src/types.ts

```diff
@@ -109,7 +109,8 @@ export type AnyIterableIterator<T> = AnyIterable<T> | AnyIterator<T>;
  * Pipe-through type (return type for all operators)
  */
 export interface Operation<T, R> {
-    (i: AnyIterable<T>): Operation<T, R>;
+    (i: Iterable<T>): Iterable<R>;
+    (i: AsyncIterable<T>): AsyncIterable<R>;
```

Comment:
This is specifying that an operation must support both sync and async behaviors.
Instead it should state that it must support at least one.

This could possible be the definition:

```ts
export type Operation<T, R> = {
    (i: Iterable<T>): Iterable<R>;
} | {
    (i: AsyncIterable<T>): AsyncIterable<R>;
};
```

Though it would probably be better to break that down further into `OperationSync`, `OperationAsync` and `AnyOperation`.

Reply:
@RebeccaStevens,

> Though it would probably be better to break that down further into OperationSync, OperationAsync and AnyOperation.

Along the same lines, I had made the following change, but rejected it to keep the diff smaller:
```ts
export type Operation<T, R>  = OperationSync<T, R> | OperationAsync<T, R>;

export interface OperationSync<T, R> {
    (i: Iterable<T>): Iterable<R>;
}

export interface OperationAsync<T, R> {
    (i: AsyncIterable<T>): AsyncIterable<R>;
}
```